articles/azure-monitor/insights/container-insights-faq.md

```diff
@@ -9,6 +9,46 @@ ms.date: 10/15/2019
 
 This Microsoft FAQ is a list of commonly asked questions about Azure Monitor for containers. If you have any additional questions about the solution, go to the [discussion forum](https://feedback.azure.com/forums/34192--general-feedback) and post your questions. When a question is frequently asked, we add it to this article so that it can be found quickly and easily.
 
+## I dont see 'Image' and 'Name' fields populated for my logs in ContainerLog table
+
+Starting with the agent version ciprod12042019 (and later) , by default, these two fields are not populated for every logline, to save customer costs on log data collected. There are two options to get these field values -
+
+- option-1 
+
+Join other tables to get these field values:
+
+Please alter your queries to get 'Image' and 'ImageTag' fields from 'ContainerInventory' table by joining on 'ContainerID' field. You can get the 'Name' field (as it used to appear in ContainerLog table) from 'KubepodInventory' table's 'ContainerName' field by joining on 'ContainerID' field. This is the recommended option.
+
+The following example is a sample detailed query that explains how to get these field values with joins
+
+```
+//lets say we are querying an hour worth of logs
+let startTime = ago(1h);
+let endTime = now();
+//below gets the latest Image & ImageTag for every containerID, during the time window
+let ContainerInv = ContainerInventory | where TimeGenerated >= startTime and TimeGenerated < endTime | summarize arg_max(TimeGenerated, *)  by ContainerID, Image, ImageTag | project-away TimeGenerated | project ContainerID1=ContainerID, Image1=Image ,ImageTag1=ImageTag;
+//below gets the latest Name for every containerID, during the time window
+let KubePodInv  = KubePodInventory | where ContainerID != "" | where TimeGenerated >= startTime | where TimeGenerated < endTime | summarize arg_max(TimeGenerated, *)  by ContainerID2 = ContainerID, Name1=ContainerName | project ContainerID2 , Name1;
+//now join the above 2 to get a 'jointed table' that has name, image & imagetag. Outer left is safer in-case there are no kubepod records are if they are latent
+let ContainerData = ContainerInv | join kind=leftouter (KubePodInv) on $left.ContainerID1 == $right.ContainerID2;
+//now join ContainerLog table with the 'jointed table' above and project-away redundant fields/columns and rename columns that were re-written
+//Outer left is safer so you dont lose logs even if we cannot find container metadata for loglines (due to latency, time skew between data types etc...)
+ContainerLog
+| where TimeGenerated >= startTime and TimeGenerated < endTime 
+| join kind= leftouter (
+   ContainerData
+) on $left.ContainerID == $right.ContainerID2 | project-away ContainerID1, ContainerID2, Name, Image, ImageTag | project-rename Name = Name1, Image=Image1, ImageTag=ImageTag1 
+
+```
+
+- option-2
+
+Re-enable collection for these fields for every container logline:

```

Comment:
Reword to - Re-enable collection for these properties for every container log line.